### ISSUE TITLE
[MRESOLVER-585] Is unclear how to use custom LocalRepositoryManager with builder

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-resolver#1257]
**[Tamas Cservenak](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=cstamas)** opened **[MRESOLVER-585](https://issues.apache.org/jira/browse/MRESOLVER-585?redirect=false)** and commented

We have a bit of chicken-and-egg problem here, as LocalRepositoryManager is created by system and one needs session for it, but session cannot be created without LocalRepositoryManager.

Session Builder provides "workaround" in form of withXXX methods, but those all cannot use custom LocalRepositoryManager...

In short: this method is "special". The usual way to use custom LRM is to integrate it with LRMProvider and tune it with priorities and/or custom LocalReposirory type strings...

---

**Affects:** 2.0.0

**Remote Links:**
- [GitHub Pull Request #548
](https://github.com/apache/maven-resolver/pull/548)
